src/Http/ServerRequest.php

```diff
@@ -517,6 +539,34 @@ public function is(array|string $type, mixed ...$args): bool
         return $this->_detectorCache[$type] = $this->_detectorCache[$type] ?? $this->_is($type, $args);
     }
 
+    /**
+     * Returns a piece of data from the Request converted to a given data type.
+     *
+     * Example: `asType('Int', $this->params, 'id', 0)` will return
+     * the 'id' parameter from the request converted to an integer.
+     *
+     * @param string $type The desired type to convert the data to. Example: 'Int', 'Bool', 'String'.
+     * @param object|array $source Source from which data should be extracted.
+     * @param mixed ...$args List of arguments
+     * @return mixed The requested data converted to the specified type.
+     * @throws \InvalidArgumentException If no detector has been set for the provided type.
+     */
+    public function asType(string $type, array|object $source, mixed ...$args): mixed
```

Comment:
This doesn't seem to be connected to ServerRequest.

Reply:
One of the benefits of the standalone functions for casting was that their return types made the output typed. For example `function toBool(mixed $value): ?bool` has a concrete return type that application code can then depend on. If our casting functions return `mixed` we haven't really improved the typesafety of the application code as any usage of the return of `toType()` will still need to do typechecks/guards as the return is mixed.

I know there was some discussion in discord about having global functions and not wanting to import functions. We could put all of the casting functions into a static class like `Hash` or `Security`. My personal preference is to just have functions as they are simpler than a static class and now that we have function imports, and namespaces we don't have to worry about polluting global namesspaces.

Reply:
Deleted my previous comment. I had totally misunderstood the intent of this change when I made it.